pygsti/circuits/circuit.py

```diff
@@ -840,9 +840,38 @@ def __radd__(self, x):
         return x.__add__(self)
 
     def __add__(self, x):
+        """
+        Method for adding circuits, or labels to circuits.
+
+        Parameters
+        ----------
+        x : `Circuit` or tuple of `Label` objects
+            `Circuit` to add to this `Circuit`, or a tuple of Labels to add to this
+            Circuit. Note: If `x` is a `Circuit` it must have line labels that are
+            compatible with this it is being added to. In other words, if `x` uses
+            the default '*' placeholder as its line label and this Circuit does not,
+            and vice versa, a ValueError will be raised.
+
+        Returns
+        -------
+        Circuit
+        """
+
         if not isinstance(x, Circuit):
             assert(all([isinstance(l, _Label) for l in x])), "Only Circuits and Label-tuples can be added to Circuits!"
             return Circuit._fastinit(self.layertup + x, self.line_labels, editable=False)
+        
+        #check that the line labels are compatible between circuits.
+        #i.e. raise error if adding circuit with * line label to one with
+        #standard line labels.
+        if (x.line_labels == ('*',) and self.line_labels !=('*',)) or (x.line_labels != ('*',) and self.line_labels ==('*',)):
+            raise ValueError("Adding circuits with incompatible line labels. This likely means that one of the circuits being"\
+                             +" added has a line label of '*' while the other circuit does not. The '*' line label is a placeholder"\
```

Comment:
I suggest
```python
msg = f"Adding circuits with incompatible line labels: {combined_labels}."  \
 + "The problem is that one of these labels uses the placeholder value of '*', while the other label does not."\
 + "The placeholder value arises when ... " 
```

Reply:
Thanks for the suggestion, I've made this change.